extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -192,11 +179,23 @@ export class AmpAdApiHandler {
    * @private
    */
   updateSize_(height, width, source, origin) {
-    this.baseInstance_.attemptChangeSize(height, width).then(() => {
-      this.sendEmbedSizeResponse_(
-        true /* success */, width, height, source, origin);
-    }, () => this.sendEmbedSizeResponse_(
-        false /* success */, width, height, source, origin));
+    let newHeight, newWidth;
```

Comment:
Can/should code be shared with `amp-iframe`?

Now that you know the code, can you add some docs about the intention (here and in `amp-iframe`? Will make refactoring easier.